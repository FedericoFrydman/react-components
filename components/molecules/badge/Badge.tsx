import { forwardRef, HTMLProps, Ref } from 'react';
import cn from 'classnames';

import { Text } from '../../atoms/text/Text';
import './badge.scss';

export type BadgeVariant = 'default' | 'success' | 'attention' | 'important' | 'new' | 'primary';

export interface BadgeProps extends Omit<HTMLProps<HTMLDivElement>, 'css'> {
  /**
   * Choose from 5 style variants. Default: 'default'.
   */
  variant?: BadgeVariant;
  /**
   * Use bold style. Default: false
   */
  bold?: boolean;
  /**
   * Use the circular badge to indicate a count of items related to an element.
   */
  circle?: boolean;
  /**
   * Text to display
   */
  text: string;
  /**
   * Use small to use the smaller 48x15 variant of badges
   */
  small?: boolean;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * The ref to the HTML DOM element
   */
  ref?: Ref<HTMLDivElement>;
}

const BadgeComponent = (
  { variant = 'default', bold = false, text, classNames = [], circle = false, small = false, ...props }: BadgeProps,
  ref: BadgeProps['ref'],
) => {
  const classBadge = 'lsux-badge';
  const cnBadge = cn(
    {
      [`${classBadge}`]: true,
      [`${classBadge}--${variant}--bold`]: bold,
      [`${classBadge}--${variant}--subtle`]: !bold,
      [`${classBadge}--circle`]: circle,
      [`${classBadge}--square`]: !circle,
      [`${classBadge}--small`]: small,
    },
    classNames,
  );

  return (
    <div {...props} className={cnBadge} ref={ref}>
      <Text as="span" text={text} textSize={small ? 'small' : 'medium'} />
    </div>
  );
};

export const Badge = forwardRef<HTMLDivElement, BadgeProps>(BadgeComponent);
