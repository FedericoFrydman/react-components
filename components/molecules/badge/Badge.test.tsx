import { shallow, ShallowWrapper } from 'enzyme';
import { Badge } from './Badge';

describe('Badge Component', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Badge text="something" />);
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
