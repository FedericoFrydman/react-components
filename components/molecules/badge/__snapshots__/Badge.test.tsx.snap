// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component should render the component 1`] = `
<div
  className="lsux-badge lsux-badge--default--subtle lsux-badge--square"
>
  <ForwardRef(TextComponent)
    as="span"
    text="something"
    textSize="medium"
  />
</div>
`;
